cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java

```diff
@@ -142,4 +165,132 @@ public void setMissingIncludeWarningsEnabled(boolean enabled){
   public boolean getMissingIncludeWarningsEnabled(){
     return this.missingIncludeWarningsEnabled;
   }
+  
+  public void setCompilationPropertiesWithBuildLog(List<File> reports, String fileFormat, String charsetName) {
+    
+    if(reports == null) {
+      return;
+    }
+    
+    for(File buildLog : reports) {
+      if (buildLog.exists()) {
+        LOG.debug("Parse build log  file '{}'", buildLog.getAbsolutePath());
+        if (fileFormat.equals("Visual C++")) {
+          parseVCppLog(buildLog, charsetName);
+        }
+
+        LOG.debug("Parse build log OK: includes: '{}' defines: '{}'", uniqueIncludes.size(), uniqueDefines.size());
+      } else {
+        LOG.error("Compilation log not found: '{}'", buildLog.getAbsolutePath());
+      }    
+    }
+  }
+
+  private void parseVCppLog(File buildLog, String charsetName) {
+
+      try {
+        BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(buildLog), charsetName));
+        String line;
+        String currentProjectPath = "";
+        while ((line = br.readLine()) != null) {
+          if (line.startsWith("  INCLUDE=")) { // handle environment includes 
+            String[] includes = line.split("=")[1].split(";");
+            for(String include : includes) {
+              if (!uniqueIncludes.contains(include)) {
+                uniqueIncludes.add(include);
+              }
+            }
+          }
+          
+          // get base path of project to make 
+          // Target "ClCompile" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppCommon.targets" from project "D:\Development\SonarQube\cxx\sonar-cxx\integration-tests\testdata\googletest_bullseye_vs_project\PathHandling.Test\PathHandling.Test.vcxproj" (target "_ClCompile" depends on it):
+          if (line.startsWith("Target \"ClCompile\" in file")) {
+            currentProjectPath = line.split("\" from project \"")[1].split("\\s+")[0].replace("\"", "");              
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\BullseyeCoverage\\bin\\CL.exe")) {
```

Comment:
Dont know of other coverage tools that intercept the compiler the way
bullseye does. But once users use this they will raise issues and we can
support other interceptors

On Thu, Apr 9, 2015, 21:38 Günter Wirth <notifications@github.com> wrote:

> In cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java
> <https://github.com/wenns/sonar-cxx/pull/399#discussion_r28089710>:
>
> > +          if (line.startsWith("  INCLUDE=")) { // handle environment includes
> > +            String[] includes = line.split("=")[1].split(";");
> > +            for(String include : includes) {
> > +              if (!uniqueIncludes.contains(include)) {
> > +                uniqueIncludes.add(include);
> > +              }
> > +            }
> > +          }
> > +
> > +          // get base path of project to make
> > +          // Target "ClCompile" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppCommon.targets" from project "D:\Development\SonarQube\cxx\sonar-cxx\integration-tests\testdata\googletest_bullseye_vs_project\PathHandling.Test\PathHandling.Test.vcxproj" (target "_ClCompile" depends on it):
> > +          if (line.startsWith("Target \"ClCompile\" in file")) {
> > +            currentProjectPath = line.split("\" from project \"")[1].split("\\s+")[0].replace("\"", "");
> > +          }
> > +
> > +          if (line.contains("C:\\Program Files (x86)\\BullseyeCoverage\\bin\\CL.exe")) {
>
> Call via BullsEyeCoverage? What if a other tool is used to create coverage?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wenns/sonar-cxx/pull/399/files#r28089710>.
>